File: 13_Linked_List_Problems/12_memory_efficient_DLL.py
class Node:
	def __init__(self, data, next = None, prev = None):
		self.data = data
		self.next = next
		self.prev = prev

class EfficientDLL:
	def __init__(self):
		self.head = None
		self.tail = None

	def insertAtEnd(self, data):
		newNode = Node(data)
		if self.head:
			

File: 13_Linked_List_Problems/12_memory_efficient_DLL.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node
{
	int data;
	struct Node *next;
	struct Node *prev;
}Node;

typedef struct DoublyLinkedList
{
	Node *head;
	Node *tail;
	int memory_effiecient;
}DoublyLinkedList;

void insert_at_end(Node *head, Node *tail, int data)
{
	Node newNode = {data, NULL, NULL}
	if tail
	{
		tail->next
	}
	else
	{
		head = &newNode;
		tail = &newNode;
	}
}

void print_it(Node *head)
{
	// 
}

void make_memory_efficient(Node *head){
	// 
}

int main(int argc, char const *argv[])
{
	DoublyLinkedList dll, *d;
	d = &dll;
	d->head = NULL;
	d->tail = NULL;

	for (int i = 1; i <= 10; ++i)
	{
		insert_at_end(d->tail, i)
	}
	return 0;
}